app/common/directives/role-selector.directive.js

```diff
@@ -0,0 +1,32 @@
+module.exports = RoleSelectorDirective;
+
+RoleSelectorDirective.$inject = [];
+
+function RoleSelectorDirective() {
+    return {
+        restrict: 'E',
+        scope: {
+            model: '=',
+            title: '='
+        },
+        controller: RoleSelectorController,
+        template: require('./role-selector.html')
+    };
+}
+RoleSelectorController.$inject = ['$scope', 'RoleEndpoint'];
+
+function RoleSelectorController($scope, RoleEndpoint) {
+    $scope.addAllRoles = addAllRoles;
+
+    // getting available roles from api
+    RoleEndpoint.query().$promise.then(function (roles) {
+        $scope.roles = roles;
+    });
+
+    // adding all available roles to model if user clicks 'Everyone'
+    function addAllRoles() {
```

Comment:
Here, do we want to set the everyone_can_see property of the category? Was the original behaviour to add all the roles in an array if so we can probably keep that.